src/pages/index.js

```diff
@@ -44,46 +49,149 @@ popupWithImage.setEventListeners();
+    "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeCard
+  );
   return card.createCardElement();
 };
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      section.addItem(cardElement);
-    },
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co",
+  headers: {
+    authorization: "35337f3b-35e8-4dc0-a9b5-b6c4dd4127c3",
+    "Content-Type": "application/json",
   },
-  ".gallery__cards"
-);
+});
 
-section.renderItems();
+let cardList;
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  pictureSelector: ".profile__image",
 });
 
+let userId;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([user, cards]) => {
+    // userInfo.setProfileInfo(userData.name, userData.about);
+    userInfo.updateUserInfo({ name: user.name, job: user.about });
+    userInfo.updateProfileAvatar(user.avatar);
+    userId = user._id; // set user id
+
+    const section = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          const cardElement = createCard(item, userId);
+          section.addItem(cardElement);
+        },
+      },
+      ".gallery__cards"
+    );
+
+    section.renderItems();
+    cardList = section;
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 const handleProfileFormSubmit = (e) => {
   e.preventDefault();
   // profileTitle.textContent = profileNameInput.value;
   // profileDescription.textContent = profileDescriptionInput.value;
   // editProfileFormValidator.disableSubmitButton();
   const inputValues = profileEditPopup.getInputValues();
-  userInfo.setUserInfo({
-    name: inputValues.Name,
-    job: inputValues.Title,
-  });
+  // userInfo.setUserInfo({
+  //   name: inputValues.Name,
+  //   job: inputValues.Title,
+  // });
+  toggleSaving("#profile-edit-modal");
+  api
+    .updateUserInfo({ name: inputValues.Name, about: inputValues.Title })
+    .then((result) => {
+      userInfo.updateUserInfo({ name: result.name, job: result.about });
+      profileEditPopup.close();
+    })
+    .finally(() => {
+      toggleSaving("#profile-edit-modal");
+    });
 };
 
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit
 );
+console.log(profileEditPopup);
+//new code
+
+const handleNewAvatarSubmit = (e) => {
+  e.preventDefault();
+  // profileTitle.textContent = profileNameInput.value;
+  // profileDescription.textContent = profileDescriptionInput.value;
+  // editProfileFormValidator.disableSubmitButton();
+  const inputValues = updateUserAvatar.getInputValues();
+  // userInfo.setUserInfo({
+  //   name: inputValues.Name,
+  //   job: inputValues.Title,
+  // });
+  // console.log(inputValues);
+  toggleSaving("#profile-change-modal");
+  api.updateUserAvatar(inputValues.Name).then((result) => {
+    toggleSaving("#profile-change-modal");
```

Comment:
Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Please fix that for every request